rotkehlchen/tests/unit/test_cost_basis.py

```diff
@@ -785,139 +815,150 @@ def test_missing_acquisitions(accountant):
-    )
-    event4 = AssetAcquisitionEvent(
-        amount=FVal(1),
-        timestamp=4,
-        rate=FVal(200),
-        index=4,
-    )
-    asset_events.acquisitions_manager.add_acquisition(event3)
-    assert asset_events.acquisitions_manager.remaining_amount == FVal(1)
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(100)
-    asset_events.acquisitions_manager.add_acquisition(event4)
-    assert asset_events.acquisitions_manager.remaining_amount == FVal(2)
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(300)
-    cost_basis_result = asset_events.acquisitions_manager.calculate_spend_cost_basis(
-        spending_amount=FVal(0.5),
-        spending_asset=asset,
-        timestamp=0,
-        missing_acquisitions=cost_basis.missing_acquisitions,
-        used_acquisitions=asset_events.used_acquisitions,
-        settings=cost_basis.settings,
-        timestamp_to_date=cost_basis.timestamp_to_date,
-        average_cost_basis=asset_events.acquisitions_manager.current_average_cost_basis,
-    )
-    assert asset_events.acquisitions_manager.remaining_amount == FVal(1.5)
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(225)
-    assert cost_basis_result.is_complete is True
-    assert cost_basis_result.taxable_bought_cost == FVal(225)
-    asset_events.acquisitions_manager.add_acquisition(AssetAcquisitionEvent(
-        amount=FVal(0.5),
-        timestamp=5,
-        rate=FVal(500),
-        index=5,
-    ))
-    assert asset_events.acquisitions_manager.remaining_amount == FVal(2)
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(475)
-
-    # see that using more than the available acquisitions adds a MissingAcquisition
-    assert asset_events.acquisitions_manager.calculate_spend_cost_basis(
-        spending_amount=FVal(3.5),
-        spending_asset=asset,
-        timestamp=0,
-        missing_acquisitions=cost_basis.missing_acquisitions,
-        used_acquisitions=asset_events.used_acquisitions,
-        settings=cost_basis.settings,
-        timestamp_to_date=cost_basis.timestamp_to_date,
-        average_cost_basis=asset_events.acquisitions_manager.current_average_cost_basis,
-    ).is_complete is False
-    assert asset_events.acquisitions_manager.remaining_amount == ZERO
+    asset_events = cost_basis.get_events(A_ETH)
+    manager = asset_events.acquisitions_manager
+
+    add_acquisition(pot, ONE, price=FVal(100))
+    assert manager.remaining_amount == FVal(1)
+    assert manager.total_value == FVal(100)
+    add_acquisition(pot, ONE, price=FVal(200))
+    assert manager.remaining_amount == FVal(2)
+    assert manager.total_value == FVal(300)
+    add_spend(pot, FVal(0.5))  # Spend only from one acquisition
+    last_cost_basis = pot.processed_events[-1].cost_basis
+    assert manager.remaining_amount == FVal(1.5)
+    assert manager.total_value == FVal(225)
+    assert last_cost_basis.is_complete is True
```

Comment:
Hmmm I don't think this is a good idea, since if the values in the events will be produced incorrectly by the cost basis calculation, there will be a good chance that we will not notice it.

I would say that in the tests is better to be as explicit as possible

Reply:
Having the events is probably better as we tried to do this in other tests too to make it more explicit. Constant values are harder to understand where they come from. But if it's just 2-3 events then it makes more sense to see that, say it's `event2.amount - event1.amount` versus `150-100`.